src/app/branded/step-1/branded-checkout-step-1.component.js

```diff
@@ -154,12 +159,116 @@ class BrandedCheckoutStep1Controller {
   checkSuccessfulSubmission () {
     if (every(this.submission, 'completed')) {
       if (every(this.submission, { error: false })) {
-        this.next()
+        if (this.useV3 === 'true') {
+          this.submitOrderInternal()
+        } else {
+          this.next()
+        }
       } else {
         this.submitted = false
       }
     }
   }
+
+  loadCart () {
+    this.errorLoadingCart = false
+
+    // Return the observable instead of subscribing here
+    return this.cartService.get().pipe(
+      tap(data => {
+        // Setting cart data and analytics
+        this.cartData = data
+        this.brandedAnalyticsFactory.saveCoverFees(this.orderService.retrieveCoverFeeDecision())
+        this.brandedAnalyticsFactory.saveItem(this.cartData.items[0])
+        this.brandedAnalyticsFactory.addPaymentInfo()
+        this.brandedAnalyticsFactory.reviewOrder()
+        console.log('done loading cart')
+      }),
+      catchError(error => {
+        // Handle errors by setting flag and logging the error
+        this.errorLoadingCart = true
+        this.$log.error('Error loading cart data for branded checkout step 2', error)
+        return Observable.throw(error) // Rethrow the error so the observable chain can handle it
+      })
+    )
+  }
+
+  loadCurrentPayment () {
+    this.loadingCurrentPayment = true
+
+    return this.orderService.getCurrentPayment().pipe(
```

Comment:
This can use the same "save observable to a variable, subscribe to it, and return it" pattern.